sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainerResponse.java

```diff
@@ -12,7 +12,7 @@
     private final CosmosContainer container;
 
     CosmosContainerResponse(CosmosAsyncContainerResponse response, CosmosDatabase database, CosmosClient client) {
-        super(response.getProperties());
+        super(response.resourceResponseWrapper, response.getProperties());
```

Comment:
Won't sending wrapper out  break design abstractions?

Reply:
wrapper is already being used in all the public API classes as to fetch the properties. This won't break out the abstraction design. 
This will just make sure to assign the wrapper in the parent class from the response object.